src/components/screens/wallet/overview/accountInfo/identity.js

```diff
@@ -1,54 +1,116 @@
+          onClick={onClickFn}
+        >
+          {truncateAddress(newAddress)}
+        </span>
+      )}
+    >
+      <span>{t('Click to see full and old addresses')}</span>
+    </Tooltip>
+    <Tooltip
+      className={`${styles.tooltip} ${styles.secondaryTooltip} ${styles.full}`}
+      content={(
+        <span
+          className={`${styles.text} ${styles.noSelect} account-secondary-full`}
+          onClick={onClickFn}
+        >
+          {newAddress}
+        </span>
+      )}
+    >
+      <span>{t('Click to hide full and old addresses')}</span>
+    </Tooltip>
+    <Legacy legacyAddress={legacyAddress} />
+  </div>
+);
+
+const WithOutTitle = ({
+  newAddress,
+  legacyAddress,
+  t = str => str,
+  onClickFn,
+}) => (
+  <div className={styles.identity}>
+    <Tooltip
+      className={`${styles.tooltip} ${styles.primary} ${styles.truncated}`}
+      content={(
+        <span
+          className={`${styles.text} ${styles.primary} ${styles.noSelect} account-primary-truncated`}
+          onClick={onClickFn}
+        >
+          {truncateAddress(newAddress)}
+        </span>
+      )}
+    >
+      <span>{t('Click to see full and old addresses')}</span>
+    </Tooltip>
+    <Tooltip
+      className={`${styles.tooltip} ${styles.primaryTooltip} ${styles.full}`}
+      content={(
+        <span
+          className={`${styles.text} ${styles.primary} ${styles.noSelect} delegate-primary-full`}
+          onClick={onClickFn}
+        >
+          {newAddress}
+        </span>
+      )}
+    >
+      <span>{t('Click to hide full and old addresses')}</span>
+    </Tooltip>
+    <Legacy legacyAddress={legacyAddress} />
+  </div>
+);
+
 /**
  *
  * Displays address and username
  * The address is initially truncated but
  * but toggles expanded on user click
  *
  */
-const Identity = ({
-  address,
-  bookmark,
-  username = '',
-}) => {
-  if (!address) return null;
-  const truncatedAddress = truncateAddress(address);
-  const [isTruncated, setTruncationState] = useState(true);
-  const onClick = () => setTruncationState(!isTruncated);
-
-  return (
-    <div className={styles.text}>
-      {
-        username || !!bookmark
-          ? (
-            <>
-              <span className={`${styles.primary} account-primary`}>
-                {username || bookmark.title}
-              </span>
-              <span
-                className={`${styles.secondary} ${styles.noSelect} delegate-secondary`}
-                onClick={onClick}
-              >
-                {isTruncated ? truncatedAddress : address}
-              </span>
-            </>
-          )
-          : (
-            <span
-              className={`${styles.primary} ${styles.noSelect} account-primary`}
-              onClick={onClick}
-            >
-              {isTruncated ? truncatedAddress : address}
-            </span>
-          )
-      }
-    </div>
-  );
+const Identity = (props) => {
+  if (!props.newAddress) {
```

Comment:
This check should be done where the component is used, not inside the component.

Reply:
Merged the components.